spec/data-types-and-values.html

```diff
@@ -170,38 +169,40 @@ <h1>
         </h1>
         <dl class="header"></dl>
         <emu-alg>
-          1. Let _valueString_ be ? ToString(_argument_.[[Value]]).
+          1. Let _value_ be ? Unbox(_argument_).
+          1. Let _valueString_ be ? ToString(_value_).
           1. Return the string-concatenation of *"Box("*, _valueString_, and *")"*.
         </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-ecmascript-language-types-box-samevalue" type="abstract operation">
+      <emu-clause id="sec-ecmascript-language-types-box-unbox" type="abstract operation">
         <h1>
-          BoxSameValue (
-            _x_: a Box,
-            _y_: a Box,
+          Unbox (
+            _box_: a Box,
           )
         </h1>
-        <dl class="header"></dl>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It extracts the boxed Object from _box_, throwing if it has been boxed by a different Realm's Box constructor.</dd>
+        </dl>
         <emu-alg>
-          1. Let _xValue_ be _x_.[[Value]].
-          1. Let _yValue_ be _y_.[[Value]].
-          1. Return ! SameValue(_xValue_, _yValue_.)
+          1. Let _realm_ be the current Realm Record.

```

Comment:
I'm still unclear how realms are threaded through the spec, but I believe we settled on comparing the realm stored by the box with the realm which created the `Box.unbox` intrinsic. This reads to me that the operation uses the current executing realm.

This is an important distinction to avoid making `Box.unbox` sensitive to the calling realm.

In particular:

```js
const otherUnbox = makeLegacyRealm().globalThis.Box.unbox;
const box = Box({});

assert.throws(() => otherUnbox(box));
```

Reply:
I believe built-in functions are identical to normal functions when it comes to realms handling.

When you evaluate `otherUnbox(box)`, steps 4-5 of [`PrepareForOrdinaryCall`](https://tc39.es/ecma262/#sec-prepareforordinarycall) sets the "current execution context" (which stores the "current Realm") to the realm associated to the `unbox` function.

Reply:
My bad, I confused execution context and environment record